File: todo_frontend/lib/models/todo_model.dart
class Todo {
  final String id;
  final String title;
  final String description;
  final bool completed;

  Todo({
    required this.id,
    required this.title,
    this.description = '',
    this.completed = false,
  });

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['_id'],
      title: json['title'],
      description: json['description'],
      completed: json['completed'],
    );
  }
}

File: backend/models/todoModel.js
const mongoose = require('mongoose');

const todoSchema = new mongoose.Schema({
  user_id: { type: mongoose.Schema.Types.ObjectId,
     required: true,
     ref: 'User' },

  title: { type: String, required: true },
  description: { type: String },
  completed: { type: Boolean, default: false },
},

{
    
timestamps: true
}
);
module.exports = mongoose.model('Todo', todoSchema);